packages/vite/native-import.js

```diff
@@ -0,0 +1,9 @@
+/**
+ * Typescript automatically compiles dynamic `import()` to `resolve()`
+ * so this needs to be a raw JS file until `typescript@4.5.0` lands with
+ * `module: "node12"` support.
+ *
+ * @param {string} fileUrl
+ * @returns {any}
+ */
+module.exports = (fileUrl) => import(fileUrl);
```

Comment:
Alternatively, we could do this in `ssrModuleLoader.ts`:
```ts
const nativeImport = eval('u => import(u)')
```

Reply:
Can we? IIRC `import()` is not an actual function, but a keyword, so I wouldn't expect this `eval` to work. Happy to be wrong, though! If this works and is more desirable I'll gladly update it.

Also worth noting again that this is a temporary workaround that will be unblocked by `typescript@4.5.0`'s `node12` setting which can preserve `import()` as written.

Reply:
It's wrapping `import` in a function, so it works fine :)

Reply:
Dang, cool this totally works! Will add a note to remove when `typescript@4.5.0` drops.